kuma/users/tests/test_views.py

```diff
@@ -275,9 +282,101 @@ def test_post_banned_user(self):
         eq_(bans.first().by, self.admin)
         eq_(bans.first().reason, 'Spam')
 
-#    TODO: Phase II:
-#    def test_post_submits_revisions_to_akismet_as_spam(self):
-#
+    @override_config(AKISMET_KEY='dashboard')
+    @requests_mock.mock()
+    def test_post_submits_revisions_to_akismet_as_spam(self, mock_requests):
+        """POSTing to ban_user_and_cleanup url submits revisions to akismet."""
+        # Create 3 revisions for self.testuser, titled 'Revision 1', 'Revision 2'...
+        num_revisions = 3
+        revisions_created = self.create_revisions(
+            num=num_revisions,
+            document=self.document,
+            creator=self.testuser)
+
+        # Enable Akismet and mock calls to it
+        Flag.objects.create(name=SPAM_SUBMISSIONS_FLAG, everyone=True)
+        mock_requests.post(VERIFY_URL, content='valid')
+        mock_requests.post(SPAM_URL, content=Akismet.submission_success)
+
+        # The request
+        data = {'revision-id': [rev.id for rev in revisions_created]}
+        resp = self.client.post(self.ban_testuser_url, data=data)
+        eq_(200, resp.status_code)
+
+        # All of self.testuser's revisions have been submitted
+        testuser_submissions = RevisionAkismetSubmission.objects.filter(revision__creator=self.testuser.id)
+        eq_(testuser_submissions.count(), num_revisions)
+        for submission in testuser_submissions:
+            ok_(submission.revision in revisions_created)
+        # No revisions submitted for self.testuser2, since self.testuser2 had no revisions
+        testuser2_submissions = RevisionAkismetSubmission.objects.filter(revision__creator=self.testuser2.id)
+        eq_(testuser2_submissions.count(), 0)
+        # Akismet endpoints were called twice for each revision
+        ok_(mock_requests.called)
+        eq_(mock_requests.call_count, 2 * num_revisions)
+
+    @override_config(AKISMET_KEY='dashboard')
+    @requests_mock.mock()
+    def test_post_submits_no_revisions_to_akismet_as_spam(self, mock_requests):
+        """
+        POSTing to ban_user_and_cleanup url does not submit to akismet.
+
+        This occurs when: 1.) User has no revisions 2.) User's revisions were
+        not in request.POST (not selected in the template)  3.) User being
+        banned did not create the revisions being POSTed.
+        """
+        # Create 3 revisions for self.testuser, titled 'Revision 1', 'Revision 2'...
+        num_revisions = 3
+        revisions_created = self.create_revisions(
+            num=num_revisions,
+            document=self.document,
+            creator=self.testuser)
+
+        # Enable Akismet and mock calls to it
+        Flag.objects.create(name=SPAM_SUBMISSIONS_FLAG, everyone=True)
+        mock_requests.post(VERIFY_URL, content='valid')
+        mock_requests.post(SPAM_URL, content=Akismet.submission_success)
+
+        # Case 1.) User has no revisions
+        data = {'revision-id': []}
+
```

Comment:
These three cases should be split into three tests, which probably means creating some helper methods to reduce test duplication.